common/src/test/java/com/thoughtworks/go/domain/PipelineLabelTest.java

```diff
@@ -282,6 +280,34 @@ public void shouldReturnEmptyIfThereIsNoMatchingEnvironmentVariable() {
         assertThat(label.toString(), is("release-"));
     }
 
+    @Test
+    public void PipelineLabelTemplateMigration_shouldMigrateColonToUnderscoreForPackageMaterial() {
+        List<List<String>> expectations = Arrays.asList(
+                Arrays.asList("${COUNT}", "${COUNT}"),
+                Arrays.asList("something-${COUNT}", "something-${COUNT}"),
+                Arrays.asList("begin-${COUNT}-end", "begin-${COUNT}-end"),
+
+                Arrays.asList("${repo:package}", "${repo_package}"),
+                Arrays.asList("start-${repo:package}-end", "start-${repo_package}-end"),
+
+                Arrays.asList("release:v1-${repo:package}", "release:v1-${repo_package}"),
+                Arrays.asList("release:v1-${repo:package}:${svn}", "release:v1-${repo_package}:${svn}"),
+                Arrays.asList("release:v1-${repo:package}:${github[:7]}:${svn}", "release:v1-${repo_package}:${github[:7]}:${svn}"),
+                Arrays.asList("release:v1-${repo:package[:5]}:${github[:7]}:${svn}", "release:v1-${repo_package[:5]}:${github[:7]}:${svn}"),
+                Arrays.asList("start-release:v1-${repo:package}:${github[:7]}:${svn}-end", "start-release:v1-${repo_package}:${github[:7]}:${svn}-end"),
+
+                Arrays.asList("${env:var}", "${env:var}"),
```

Comment:
@arvindsv @maheshp 

As part of GoCD release `v18.10.0`, we've added support to specify pipeline label as Environment Variable with pattern `${env:env_var_name}`. But this pattern conflicts with package repository material with name `env`.

So lets say,

We've defined a package repository with name `env` and a package with name `my_package` and specified the label template of pipeline  as `${env:my_package}`. Even though, the package-repository material name is specified, it won't be resolved. Because, as per our implementation, we always resolve `${env:my_package}` to environment variable.

Hence, in this scenario, pipeline label would be set to empty as there is no env-var with name `my_package` is specified.

This bug is only applicable for GoCD releases from `v18.10.0` to `v19.11.0` as the pipeline label pattern for package material has changed to `${repo_package}` in GoCD `v19.12.0`

**For GoCD releases from `v18.10.0` to `v19.11.0`, we should say, for patterns `${env:something}`, `env` is a reserved GoCD keyword and will always be resolved to environment variable.**

With the above-mentioned behavior, for pipeline-label migrations containing `${env:something}`, the pattern template will be treated as environment variable pattern and will not migrate `:` to `_`.

Reply:
I understand. Makes sense to me.